homeassistant/components/light/elkm1.py

```diff
@@ -0,0 +1,58 @@
+"""
+Support for control of ElkM1 lighting (X10, UPB, etc).
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.elkm1/
+"""
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, Light)
+from homeassistant.components.elkm1 import (
+    DOMAIN as ELK_DOMAIN, ElkEntity, create_elk_entities)
+
+DEPENDENCIES = [ELK_DOMAIN]
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info):
+    """Set up the Elk light platform."""
+    elk = hass.data[ELK_DOMAIN]['elk']
```

Comment:
Please add a guard clause here that checks if discovery_info is None and return if so.

Send an empty dict as discovery_info from the component.

We only want to set up the platform via discovery.

Reply:
What does "only want to set up the platform via discovery" mean? Can I pass config info in discovery_info? Right now all config is passed in hass.data['elkm1'].

Should I fix alarm_control_panel the same way?

Reply:
If a user puts in a config section for the platform in the config yaml, home assistant will try to wet up the platform that way. We don't want it to complete then.

Yes, config can be passed via discovery_info. It must be serializable though.

Yes, add a guard clause in the alarm_control_panel platform too.